### DISCUSSION TITLE
以哪一方面作为入口去钻研本项目的代码？

### DISCUSSION THREAD

**Discussion by cuihe500** [alibaba/alibaba-rsocket-broker#232] [Q&A]
雷卷大佬，您好！
我使用Alibaba RSocket Broker已经一年多了 一直在做上层业务 使用起来非常稳定 效率也很高
我看到您之前回复的一个issuse提到目前本项目只有您一个人在维护 一个人的精力总是有限的 我非常想作为贡献者之一参与到本项目的开发之中
但是无奈水平有限 自己钻研了一部分还是效率较低 所以想向您请教一下 该从本项目的哪一方面作为入手点 或者需要补充哪一方面的知识再去了解会更好一些呢？
@linux-china 希望您能帮我解决一下这个问题 如果打扰到您的话 希望您能谅解🙏🙏🙏

**Comment by cuihe500** [comment]
顺便说一下，雷卷大佬是我接触到的第一位如此厉害的Java Programmer，我非常喜欢RSocket协议，也特别欣赏这种响应式编程的思想，可能正是因为我编码太少的原因，我接受这种反应式编程的思想很迅速，而且也能带给我非常好的代码编写体验
非常感谢雷卷大佬的开源和坚持不懈的精神，您这种精神影响我很深，所以我虽然现在水平很差，但是我仍然想为本项目贡献一些力量，哪怕是完善文档，尽可能解决一些issuse也可以，如果有需要我也可以做一些推广、测试等等边缘工作。

**Comment by linux-china** [comment]
@cuihe500 当然欢迎啊，谁都是从初级程序员过来的。 RSocket Broker架构和代码其实不复杂，可以说很简单的，可能在细节上需要专研一下。 其实最复杂的可能是Reactive的编程思想，但是这个也是不特别难理解，API和Java Streams APIs, JavaScript Array Methods，DataFrame等都有类似之处。对于初学者来说，可能从PPT，文档和Example入手比较好点，你做1-2个PPT，写一些blog或者文章等，这些都是非常有帮助的，当然你也可以向其他人介绍。

不用担心自己是新手，也不用担心现在学的太晚啦，行动起来更重要，有点鸡汤啦 :)

**Reply by cuihe500** [reply]
非常感谢您的回复！
作为新手的我，一下看过去就感觉东西很多，不知从哪里开始是最好的。
我感觉我欠缺很多的一些基础的东西，比如CloudEvent，Spring RSocket这些。不瞒您说，我昨天找了半天，也没找到服务在请求接入后是交给哪里负责的😓。
不过我会继续去研究这个的，还麻烦雷卷大佬多多帮助小弟一下😁。

**Reply by shareisall** [reply]
建议从 alibaba-broker-spring-boot-starter 的自动配置类 com.alibaba.spring.boot.rsocket.broker.RSocketBrokerAutoConfiguration 开始读源码。
